src/OrchardCore/OrchardCore.Users.Core/Services/UserService.cs

```diff
@@ -179,7 +179,7 @@ public async Task<bool> ResetPasswordAsync(string userIdentifier, string resetTo
                 return result;
             }
 
-            var user = await FindByUsernameOrEmailAsync(userIdentifier) as User;
+            var user = await _userManager.FindByEmailAsync(userIdentifier) as User;
```

Comment:
Should userIdentifier be renamed to email? I believe there is also a check above that says "A user name or email is required.", which isn't accurate anymore.